### ISSUE TITLE
Changed default branch and pouchdb stopped working

### ISSUE THREAD

**Issue by Robophil** [semantic-release/semantic-release#470]
Hi,

I changed my default branch to develop and semantic-release stopped working.
```javascript
{
  "branch": "develop"
}
```
My log
```bash
semantic-release WARN invalid config loglevel="notice"
[Travis Deploy Once]: Elect build leader among builds with Node versions: 8, 7, 6, 4.
[Travis Deploy Once]: Elect job 1 as build leader as it runs the highest node version (8).
[Travis Deploy Once]: Success at attempt 1. All 4 jobs passed.
[Travis Deploy Once]: All jobs are successful for this build!
semantic-release ERR! commits The commit the last release of this package was derived from is not in the direct history of the "develop" branch.
semantic-release ERR! commits This means semantic-release can not extract the commits between now and then.
semantic-release ERR! commits This is usually caused by force pushing, releasing from an unrelated branch, or using an already existing package name.
semantic-release ERR! commits You can recover from this error by publishing manually or restoring the commit "623866dc2f616b3883165ad92b15a30dafa8ebc4".
semantic-release ERR! pre Failed to determine new version.
semantic-release ERR! pre ENOTINHISTORY Commit not in history
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! microservice-lite@1.0.1 semantic-release: `semantic-release pre && npm publish && semantic-release post`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the microservice-lite@1.0.1 semantic-release script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
npm ERR! A complete log of this run can be found in:
npm ERR!     /home/travis/.npm/_logs/2017-10-13T04_51_49_079Z-debug.log

```

I've gone through #276 and [TROUBLESHOOTING.md](https://github.com/semantic-release/semantic-release/blob/caribou/TROUBLESHOOTING.md) and no progress so far

**Comment by pvdlg** [comment]
This is because the commit used to do the last release is not present in your develop branch, or is too far in history (by default Travis pull only the last 50 commits of your branch).

Please make sure the develop branch contains the commit used for the last release. In order to solve the problem of Travis pulling only the last 50 commits, you can use `semantic-release@8.2.0` that will handle this situation. It's currently released on the the `next` dist-tag so you would have to specify the version `^8.2.0` in your `package.json`.

**Comment by Robophil** [comment]
Hi,

I was previously using `master` with semantic release but i changed it to `develop`
```json
{
  "branch": "develop"
}
```
and that's when the issue started

**Comment by Robophil** [comment]
As stated above, i even tried pushing a tag manually but that didn't help as it kept asking for a specific commit as shown in the log file above.

```
semantic-release ERR! commits You can recover from this error by publishing manually or restoring the commit "623866dc2f616b3883165ad92b15a30dafa8ebc4".
```

**Comment by pvdlg** [comment]
Is you develop branch contains all the commit from master ? How did you created it ?
Did you try with `semantic-release` version `8.2.0` ?

**Comment by Robophil** [comment]
Develop was branched off master, and that's where I push to
By default, `semantic-release` was tracking `master` so I merged into master and a release was made.
I want a release to be made every time I merge into `develop`, that's why I changed my default branch to `develop`. I haven't been able to get `semantic release` to work after that.

I'm yet to try `8.2.0`, but i'll do that now.

**Comment by pvdlg** [comment]
> I'm yet to try 8.2.0, but i'll do that now.

Ok let us know how it goes!

**Comment by Robophil** [comment]
Yh, I did and faced the same issue.
Here's what I did that helped.

```bash
git checkout develop
git merge v1.0.1
```
After merging that tag into **develop**, it worked fine ;-) 
`git tag --merged develop` now shows `v1.0.1`.

Thanks for all the help @pvdlg
